### DISCUSSION TITLE
提个自己使用的建议：关于过滤规则的 自己构想的如果不合适还望作者不要见笑。

### DISCUSSION THREAD

**Discussion by xwndo** [jxxghp/MoviePilot#372] [建议]
目前MP和之前的NT过滤规则个人感觉有些复杂，也有些难懂原理。
感觉可以这样做，比如把 质量、分辨率、编码、音频、配音、字幕、效果 总共分7大类，每个大类里在放上各个的子分类，
让后让用户自己鼠标拖动排列顺序，匹配种子的时候根据每类用户自己排列的顺序依次往下匹配，
每个子分类除了可以排列顺序，还可以选择匹配和排除，启用就是参与匹配，禁用就是排除此项。
然后洗板规则和过滤规则同样，这样的话其实就不需要单独的洗板规则，可以设置开启或者关闭洗板，
开启默认电影或者TV的洗板后，就一直按照自己的排列规则来匹配，一直匹配到第一项规则资源为止，这样用户也可以设置一个上限，比如有些用户不需要蓝光就把蓝光禁用掉，第一选项设置为WEB-DL，这样就匹配或者洗板到WEB-DL就好。用户更改配置后，洗牌在重新根据用户的配置来匹配

 我感觉这样也比较符合PT的设计，PT刚上线的资源质量肯定是低的，这样自动匹配也符合PT。
和简化了过滤规则，用户设置也比较简单，只需要根据自己的需求来排列自己需要的影片质量。
每项单独排列也比较易懂。

构想我的配置：

  一、质量（1.蓝光原盘 2.BLURAY 3.UHD 4.REMUX 5.WEB-DL 6.其他 7.任何质量都匹配）ps:每一项都可以选择禁用掉，禁用掉就跳过此项匹配下一项，第7项也就是最后一项是个保底设置，开启了就可以保证有资源一定搜索的到。
 二、 分辨率(1.4K 2.2160p 3.1080p 4.720p禁用 5.其他 6.所有分辨率)ps：如第4项警用掉，最后一项6开启后还是能匹配到，或者也警用掉就匹配不到。
  三、编码（1.avc 2.hevc 3.h.265 4.h.264 5.其他 6.所有）
  四、配音（1.国语配音 2.英语配音 3.其他 4.所有）
 五、 字幕（1.特效字幕 2.中文字幕 3.英语字幕 4.其他字幕 5.无字幕 6.无要求）ps：可以选择警用，禁用后就不匹配，最后一项为保底
  六、效果（1.HDR 2.杜比视界 3.其他 4.无要求）ps：可以选择警用，禁用后就不匹配，最后一项为保底
  七、音频:（1.杜比 2.。。。。 3.所有或者无要求）

ps:比如按照我这个排序，会优先下载 蓝光的4K的AVC的国语配音特效字幕HDR杜比
第1的大分类比较重要，优先匹配，如匹配到蓝光就自保留蓝光的资源后再通过下面的筛选，就不需要再匹配UHD的了。

 这样每一项都依次匹配，也符合每个人的要求，因为PT吗，嘛总有些资源没有，规则也不能定死，这样既能往上高质量的匹配，也能在没资源的时候再往下匹配，退求次么，在一个这样每一项单独设置我觉的比较符合大众理解吧。
  这样也可以内置一个默认规则，就比如我这样的，按照顺序一次排列一般的小白也就不需要设置了，也能搜到资源，需要个性化在自己拖动排列下规则就好了。

补充：
1.大分类也可以选择排序，比如质量排第一就先根据质量先匹配，在匹配下面的，这样应该会符合逻辑些，有些人比较重视画质，有些人家里有好的音响，比较重视音质就先匹配音质，在从匹配到的音质里匹配下面的规则。这样应该会减轻点压力吧。
这样也不会形成循环匹配吧。
2.建议添加鼠标拖动排序，这样应该效率高些吧。
![画板 1](https://github.com/jxxghp/MoviePilot/assets/44762565/3010e265-d7e5-4c36-bbf2-5d2faf1ab5a8)

匹配逻辑可以设置一级一级的筛选匹配，这样就可以防止太多组合，比如优先级1里，匹配到第一个原盘，就只提取带原盘的种子，其他不展示，除非没有原盘，在往右匹配，一直到匹配到位置。
然后在从优先级1里筛选的原盘里在筛选优先级2里的，这种层层筛选会不会好些。

其实也可以不要层层筛选，就符合规则的全部展示，然后在根据自己的排序调整展现的优先级也行，这样会展示全部种子，只不过在重新排序，这样就不会漏掉种子

其实效果和各大站点选择RSS的时候有些像了。

![clipboard_20230819_133733](https://github.com/jxxghp/MoviePilot/assets/44762565/0363d985-4ef3-48cc-98fb-56ee3390dd9e)

**Comment by aicenluo** [comment]
你这个建议很赞，思路清晰，值得借鉴

**Comment by kingbb2001** [comment]
感觉不错哦

**Comment by lt19860** [comment]
搞成你这样，我就会设置的，现在的规则我有点懵

**Comment by lzyl42** [comment]
直接参考radarr那种对于指定规则赋予一个正/负分数然后按总和大小来判断就很好，可以很简单分级各部分，而且查询只需要对文件名计算一次总分就行不用每个规则组去匹配

**Comment by WaterColor-Tone** [comment]
支持，再增加一个设置后对应改变的配置文件，让希望自定义的人能够自己设置就好了。
